src/main/java/com/splunk/logging/AckLifecycleState.java

```diff
@@ -21,23 +21,48 @@
  */
 public class AckLifecycleState {
 
-  /**
-   * @return the sender
-   */
-  public HttpEventCollectorSender getSender() {
-    return sender;
-  }
   public enum State {
-    PRE_EVENT_POST, EVENT_POST_OK, EVENT_POST_NOT_OK, EVENT_POST_FAILURE, PRE_ACK_POLL, ACK_POLL_OK, ACK_POLL_NOT_OK, ACK_POLL_FAILURE
-  };  
+	// States tied to an EventBatch object
+    PRE_EVENT_POST,
+    EVENT_POST_OK,
+    EVENT_POST_NOT_OK,
+    EVENT_POST_FAILURE,
+    PRE_ACK_POLL,
+    ACK_POLL_OK,
+    ACK_POLL_NOT_OK,
+    ACK_POLL_FAILURE,
+
+    // States without an EventBatch object
+    HEALTH_POLL_OK,
+    HEALTH_POLL_NOT_OK,
+    HEALTH_POLL_FAILED
+  };
+
   private final State currentState;
-  private final EventBatch events;
+  private EventBatch events = null;
   private final HttpEventCollectorSender sender;
 
-  public AckLifecycleState(State currentState, EventBatch events, HttpEventCollectorSender sender) {
+  public AckLifecycleState(State currentState
+		  , EventBatch events
+		  , HttpEventCollectorSender sender) throws Exception {
+	if (events == null) {
```

Comment:
don't throw if this is a health poll state. And when you do throw, perhaps IllegalStateException